FlowCal/io.py

```diff
@@ -164,21 +164,18 @@ def read_fcs_text_segment(buf, begin, end, delim=None):
     buf.seek(begin)
     raw = buf.read((end+1)-begin)
 
-    # Some files have been found to include space characters at the end of the
-    # TEXT segment. If the delimiter is not a space character, strip spaces
-    # from the end of the segment.
-    if not delim.isspace():
-        raw = raw.rstrip()
+    # Check that the first character of the TEXT segment is equal to the
+    # delimiter.
+    if raw[0] != delim:
+        raise ValueError("segment should start with delimiter")
```

Comment:
maybe add a check that `len(raw)>0`? how does this parse an empty segment (e.g. ANALYSIS)? I think it never gets called? cause the `FCSFile` object checks the HEADER and finds offset values of 0, so it never calls `fc.io.read_fcs_text_segment()` with an empty segment? this is kinda an edge case, but would be nice if `read_fcs_text_segment()` succeeded when passed a degenerate segment (e.g. `raw=''` or `raw='//'`, resulting in empty dictionaries?) or failed gracefully when passed a degenerate malformed segment (`raw='/'`).